container/cgroupfs/hugetlb.go

```diff
@@ -0,0 +1,94 @@
+// +build linux
+
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2015 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cgroupfs
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/go-units"
+
+	"github.com/intelsdi-x/snap-plugin-collector-docker/container"
+)
+
+// HugeTlb implements StatGetter interface
+type HugeTlb struct{}
+
+// GetStats reads huge table metrics from Hugetlb Group
+func (h *HugeTlb) GetStats(stats *container.Statistics, opts container.GetStatOpt) error {
+	path, err := opts.GetStringValue("cgroup_path")
+	if err != nil {
+		return err
+	}
+
+	hugePageSizes, err := getHugePageSize()
+	if err != nil {
+		return err
+	}
+
+	hugetlbStats := container.HugetlbStats{}
+	for _, pageSize := range hugePageSizes {
+		usage, err := parseIntValue(filepath.Join(path, strings.Join([]string{"hugetlb", pageSize, "usage_in_bytes"}, ".")))
+		if err != nil {
+			return err
+		}
+
+		maxUsage, err := parseIntValue(filepath.Join(path, strings.Join([]string{"hugetlb", pageSize, "max_usage_in_bytes"}, ".")))
+		if err != nil {
+			return err
+		}
+
+		failcnt, err := parseIntValue(filepath.Join(path, strings.Join([]string{"hugetlb", pageSize, "failcnt"}, ".")))
+		if err != nil {
+			return err
+		}
+
+		hugetlbStats.Usage = usage
+		hugetlbStats.MaxUsage = maxUsage
+		hugetlbStats.Failcnt = failcnt
+
+		stats.Cgroups.HugetlbStats[pageSize] = hugetlbStats
```

Comment:
Small suggestion: consider removing `hugetlbStats` var in this case, because it's not really needed. It might be implemented in the following way:
```
stats.Cgroups.HugetlbStats[pageSize] = container.HugetlbStats{
Usage: usage,
MaxUsage: maxUsage,
Failcnt: failcnt,
}
```
but up to you - if you prefer leave it as it is now, that's ok for me too